internal/expr/expr_test.go

```diff
@@ -411,6 +411,18 @@ comment */
 string
 of three lines' AND id = $Person.*`,
 		err: `cannot parse expression: line 3, column 26: asterisk not allowed in input expression "$Person.*"`,
+	}, {
+		query: "SELECT &S[:] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax "S[:]" in output expression`,
+	}, {
+		query: "SELECT &S[0] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax "S[0]" in output expression`,
+	}, {
+		query: "SELECT &S[1:5] FROM t",
+		err:   `cannot parse expression: column 8: cannot use slice syntax "S[1:5]" in output expression`,
+	}, {
+		query: "SELECT col1 AS &S[1:5] FROM t",
+		err:   `cannot parse expression: column 16: cannot use slice syntax "S[1:5]" in output expression`,
 	}}
```

Comment:
It looks like you're missing tests for a few of the error messages in `parseSliceAccessor`. These one's only relate to output expressions.

Reply:
They are in the parse helper instead, I thought about having them in both places but I feel this way the code is less cluttered.

Reply:
My mistake for pressing add single comment instead of start review, but though I've left a comment in the review relating to this.